networkx/convert_matrix.py

```diff
@@ -655,7 +657,8 @@ def _dok_gen_triples(A):
 
     """
     for (r, c), v in A.items():
-        yield r, c, v
+        # Doing `v.tolist()` converts a NumPy scalar to the appropriate Python scalar
+        yield int(r), int(c), v.tolist()
```

Comment:
After feedback from @rossbar and careful investigation, this is the now only place in this PR where we do `.tolist()` to convert a NumPy scalar to a Python scalar--and this one can't be helped.